pkg/controller/container-runtime-config/helpers.go

```diff
@@ -933,17 +934,69 @@ func validateClusterImagePolicyWithAllowedBlockedRegistries(clusterScopePolicies
 	return nil
 }
 
-func generateSigstoreRegistriesdConfig(clusterScopePolicies map[string]signature.PolicyRequirements) ([]byte, error) {
+func generateSigstoreRegistriesdConfig(clusterScopePolicies map[string]signature.PolicyRequirements, registriesTOML []byte) ([]byte, error) {
 	if len(clusterScopePolicies) == 0 {
 		return nil, nil
 	}
 
-	registriesDockerConfig := make(map[string]dockerConfig)
-	sigstoreAttachment := dockerConfig{
-		UseSigstoreAttachments: true,
+	var (
+		err                    error
+		tmpFile                *os.File
+		registriesConf         = sysregistriesv2.V2RegistriesConf{}
+		registriesDockerConfig = make(map[string]dockerConfig)
+		sigstoreAttachment     = dockerConfig{
+			UseSigstoreAttachments: true,
+		}
+	)
+
+	if registriesTOML != nil {
+		tmpFile, err = os.CreateTemp("", "regtemp")
+		if err != nil {
+			return nil, err
+		}
+		defer func() {
+			tmpFile.Close()
+			os.Remove(tmpFile.Name())
+		}()
+
+		_, err = tmpFile.Write(registriesTOML)
+		if err != nil {
+			return nil, err
+		}
+
+		if _, err := toml.NewDecoder(bytes.NewBuffer(registriesTOML)).Decode(&registriesConf); err != nil {
+			return nil, fmt.Errorf("error decoding registries config: %w", err)
+		}
 	}
-	for scope := range clusterScopePolicies {
-		registriesDockerConfig[scope] = sigstoreAttachment
+
+	for policyScope := range clusterScopePolicies {
+		registriesDockerConfig[policyScope] = sigstoreAttachment
+		if registriesTOML == nil {
+			continue
+		}
+		parentReg, err := sysregistriesv2.FindRegistry(&types.SystemContext{SystemRegistriesConfPath: tmpFile.Name(), SystemRegistriesConfDirPath: os.DevNull}, policyScope)
+		if err != nil {
+			return nil, err
+		}
+
+		if err = addScopeMirrorsSigstoreRegistriesdConfig(registriesDockerConfig, policyScope, parentReg, sigstoreAttachment); err != nil {
+			return nil, fmt.Errorf("error adding clusterimagepolicy scope %s relevant mirrors: %w", policyScope, err)
+		}
+		for _, reg := range registriesConf.Registries {
+			scope := reg.Location
+			if reg.Prefix != "" {
+				scope = reg.Prefix
+			}
+			if runtimeutils.ScopeIsNestedInsideScope(scope, policyScope) && scope != policyScope {
+				nestedReg, err := sysregistriesv2.FindRegistry(&types.SystemContext{SystemRegistriesConfPath: tmpFile.Name()}, scope)
```

Comment:
This place also needs updating with `SystemRegistriesConfDirPath`.

(Please create a single `SystemContext` variable to share that.)